File: 20210917_PBjar_CTF/rev/qasm/solve.py
data = [6,34,82,81,107,65,66,47,66,0,54,53,120,70,81,6,40,82,14,22,65,1,10,18,45,85,28,53,47,32,39,32,102,71,57,27,120,65,70,98,23,125,32,74,2,101,104,62,60,55,4,14,89,63,117,100,125]
mod_by = 127

res = ""

sum = 0
for i in range(1, len(data) + 1):
	value = (data[-i] - sum) % mod_by
	c = value * pow(i, -1, mod_by) % mod_by
	res = chr(c) + res
	sum = (sum + value) % mod_by

print(res)

File: 2021/20210821_corCTF_2021/crypto/fibinary/solve.pl
#!/usr/bin/perl

use strict;
use warnings;

my @fib_cache = (1, 1);

sub fib {
	my $n = $_[0];
	if ($n < @fib_cache) { return $fib_cache[$n]; }
	for (my $i = @fib_cache; $i <= $n; $i++) {
		push(@fib_cache, $fib_cache[$i - 1] + $fib_cache[$i - 2]);
	}
	return $fib_cache[$n];
}

while (my $line = <STDIN>) {
	chomp($line);
	my @data = split(/\s+/, $line);
	for (my $i = 0; $i < @data; $i++) {
		my $res = 0;
		my $l = length($data[$i]);
		for (my $j = 0; $j < $l; $j++) {
			if (substr($data[$i], $j, 1) eq "1") {
				$res += &fib($l - $j - 1);
			}
		}
		print chr($res);
	}
}

print "\n";
